rpctest/node.go

```diff
@@ -201,16 +205,54 @@ func (n *node) debugf(format string, args ...interface{}) {
 	debugf(n.t, pid+format, args...)
 }
 
+// buildNode creates a new temporary directory and node and saves the location
+// to a package level variable where it is used for all tests. testNodeDirMtx
+// must be held for writes.
+func buildNode(t *testing.T, exe string) error {
+	var err error
+	testNodeDir, err = ioutil.TempDir("", "rpctestdcrdnode")
+	if err != nil {
+		return err
+	}
+	debugf(t, "node dir: %v\n", testNodeDir)
+	// Determine import path of this package.
+	_, rpctestDir, _, ok := runtime.Caller(1)
+	if !ok {
+		return fmt.Errorf("cannot get path to dcrd source code")
+	}
+	dcrdPkgPath := filepath.Join(rpctestDir, "..", "..")
```

Comment:
As is, this makes `rpctest` unusable for third party clients (including dcrlnd).

The easiest solution is probably to add a new API alternative such that external callers can manually specify the dcrd exe to use when creating the harness. 

Another alternative would be to use a package global to track the exe to use (and provide package functions to modify it) then check that when starting the node (if empty, build from source).

Here's a gist that can demonstrate the issue (hooked on go.mod to use the current latest commit for rpctest): https://gist.github.com/matheusd/b7fa8e3f5427c139297cc0770067c921

Reply:
Caller can now specify an executable. For you example:
```
func TestMy(t *testing.T) {
	net := chaincfg.SimNetParams()
	_, err := rpctest.NewWithDCRD(t, net, nil, nil, "./dcrd")
	if err != nil {
		t.Fatalf("unable to start harness: %v", err)
	}
}
```